examples/PlayVGM/PlayVGM.ino

```diff
@@ -0,0 +1,267 @@
+  printStringln(buf);
+
+  vgmFile.seek(OFFSET_GD3); //GD3 data offset
+  uint32_t relativeGd3Offset  = readUint32FromFile();
+
+  if (relativeGd3Offset == 0) {
+    Serial.println(F("No GD3 tag in file"));
+  } else {
+    //Expecting compability with GD3 1.00
+    vgmFile.seek(OFFSET_GD3 + relativeGd3Offset + 12);
+
+    printGd3Field(buf, BUF_SIZE, F("Track title:    ")); //1:  Track title
+    readGd3Field( buf, BUF_SIZE                       ); //2:  Track title (Japanese), ignore
+    printGd3Field(buf, BUF_SIZE, F("Game Name:      ")); //3:  Game name
+    readGd3Field( buf, BUF_SIZE                       ); //4:  Game name (Japanese), ignore
+    printGd3Field(buf, BUF_SIZE, F("System:         ")); //5:  System name
+    readGd3Field( buf, BUF_SIZE                       ); //6:  System name (Japanese), ignore
+    printGd3Field(buf, BUF_SIZE, F("Composer:       ")); //7:  Name of Original Track Author
+    readGd3Field( buf, BUF_SIZE                       ); //8:  Name of Original Track Author (Japanese), ignore
+    printGd3Field(buf, BUF_SIZE, F("Release:        ")); //9:  Date of game's release
+    printGd3Field(buf, BUF_SIZE, F("VGM by:         ")); //10: Name of person who converted it to a VGM file
+  }
+}
+
+void printGd3Field(byte *buf, int maxsize, const __FlashStringHelper* fieldName)
+{
+  readGd3Field(buf, maxsize);
+  Serial.print(fieldName);
+  printStringln(buf);
+}
+
+void readGd3Field(byte *p, int maxsize) {
+  //Gd3-fields use 16bit characters, zero-terminated
+  uint16_t c;
+  int pos = 0;
+  do {
+    c  = readUint16FromFile();
+    p[pos] = (byte)c; //Ignore high byte
+    if (pos < maxsize - 1) pos++;
+  } while (c != 0);
+  p[pos] = 0;
+}
+
+void printString(byte *p) {
+  while (*p) Serial.print((char)*p++);
+}
+
+void printStringln(byte *p) {
+  printString(p);
+  Serial.println("");
+}
+
+void loop() {
+  //Play until finished/error
+  uint32_t time = millis();
+  while (PlaybackStatus == PLAYBACK_PLAYING) {
+    playVgmSongYM3812(); //Fetch and execute one command
+
+    if (waitSamples) { //Delay command found, delay n samples
+      const uint16_t samplesPrSecond = 44100;
+      const double samplesPrMs = (double)samplesPrSecond / 1000; //44.1
+      uint16_t waitMs = (uint16_t)((double)waitSamples / samplesPrMs);
+
+      //Reduce delay with time spent on IO
+      uint32_t ioTime = millis() - time;
+      if (ioTime < waitMs) {
+        delay(waitMs - ioTime);
+      }
+      time = millis();
+    }
+    waitSamples = 0;
+  }
+}
+
+void unsupportedCode(byte b) {
+  uint8_t buf[50];
+  sprintf(buf, "Unsupported code 0x%02X offset 0x%08lX", b, vgmFile.position() - 1);
+  printStringln(buf);
+  PlaybackStatus = PLAYBACK_ERROR_UNSUPPORTED_CODE;
+}
+
+void playVgmSongYM3812() {
+
+  //Simple stupid implementation, will stop if reading unsupported byte
+  
+  byte code = vgmFile.read();
+
+  if (code == 0x5A) { //YM3812 (OPL2)
```

Comment:
Maybe for readability this would be more clear as a switch.
```suggestion
  switch (code) {
    case CODE_CHIP_YM3812:
      // Do things!
      break;
    case CODE_SAMPLE_DELAY_0 ... CODE_SAMPLE_DELAY_15
      // Do other things
      break;
    default:
      // Unsupported code
      break;
  }
```

Reply:
Sure, added as you suggested. I did not know about switch case ranges though, seems like a gcc-feature.